controllers/submission.go

```diff
@@ -71,3 +71,56 @@ func PostSubmission(ctx context.Context, w http.ResponseWriter, r *http.Request)
 		return http.StatusBadRequest, errors.New("Unknown submission kind.")
 	}
 }
+
+// FinalSubmission makes the last submission final.
+func FinalSubmission(ctx context.Context, w http.ResponseWriter, r *http.Request) (status int, err error) {
+	p, ok := passenger.FromContext(ctx)
+	if !ok {
+		return http.StatusUnauthorized, nil
+	}
+	if r.Method != "POST" {
+		return http.StatusMethodNotAllowed, nil
+	}
+
+	resultKey, err := datastore.DecodeKey(mux.Vars(r)["resultKey"])
+
+	if !util.HasParent(p.UserKey, resultKey) {
+		return http.StatusBadRequest, errors.New("Cannot submit answer for other users")
+	}
+
+	taskKey, err := datastore.DecodeKey(mux.Vars(r)["taskKey"])
+	var result model.Result
+	if err = datastore.Get(ctx, resultKey, &result); err != nil {
+		return http.StatusInternalServerError, err
+	}
+	var challenge model.Challenge
+	if err = datastore.Get(ctx, result.Challenge, &challenge); err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	var submissionKeys []*datastore.Key
+	if submissionKeys, err = model.NewQueryForCodeSubmission().
+		Ancestor(resultKey).
+		Order("-Start").
```

Comment:
Time just exists there in the entity. It is not set anywhere. I`ll handle that in this commit.